be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -326,6 +327,13 @@ Status SegmentIterator::_init_impl(const StorageReadOptions& opts) {
             if (storage_type == nullptr) {
                 storage_type = vectorized::DataTypeFactory::instance().create_data_type(*col);
             }
+
+            bool exist = _segment->column_exists(Segment::ColumnIdentifier {
```

Comment:
if the column not exist, the iterator should be `DefaultValueColumnIterator` maybe you can use this to find out if the column exist or not